src/sentry/tsdb/base.py

```diff
@@ -171,6 +172,19 @@ def calculate_expiry(self, rollup, samples, timestamp):
         epoch = self.normalize_to_epoch(timestamp, rollup)
         return epoch + (rollup * samples)
 
+    def get_earliest_timestamp(self, rollup, timestamp=None):
+        """
+        Calculate the earliest available timestamp for a rollup.
+        """
+        if timestamp is None:
+            timestamp = timezone.now()
+
+        lifespan = timedelta(seconds=rollup * (self.rollups[rollup] - 1))
+        return self.normalize_to_epoch(
+            timestamp - lifespan,
+            rollup,
+        )
```

Comment:
I was tempted to unilaterally apply this to the output of `get_range` but I'm pretty sure if I did that it'd break something. This also assumes that the clocks between Redis and the querying process aren't too far out of sync.